core/storage/ipfs/datastore.hpp

```diff
@@ -51,8 +53,39 @@ namespace fc::storage::ipfs {
      * @return success if removed or didn't exist, error otherwise
      */
     virtual outcome::result<void> remove(const CID &key) = 0;
-  };
 
+    /** Compute CID from bytes */
+    inline static outcome::result<CID> cid(gsl::span<const uint8_t> bytes) {
```

Comment:
I don't like an idea to put cid converters into datastore interface. Datastore is a simple and clean interface representing underlying storage. With cid converters it turns into datastore-with-cid-converters interface. Shortly speaking, please move cid-related method to CID or separate interface and keep interfaces clean.

Reply:
No. Go implementation provides such interface.
Template type T is like go "interface{}".